galasa-ui/src/components/test-runs/CustomCheckBoxList.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+"use client";
+import styles from "@/styles/TestRunsPage.module.css";
+import { 
+  Checkbox,
+  CheckboxGroup 
+} from "@carbon/react";
+import { Button } from "@carbon/react";
+import { useTranslations } from "next-intl";
+import { FormEvent } from "react";
+
+interface CheckBoxListProps {
+    title: string;
+    items: string[];
+    selectedItems: string[];
+    onChange: (newSelectedItems: string[]) => void;
+    onSubmit: (e: FormEvent) => void;
+    onCancel: () => void;
+  }
+  
+/**
+ * CheckBoxList component for displaying a list of checkboxes.
+ * @param title - The title of the checkbox list.
+ * @param items - The list of items to display as checkboxes.
+ * @param selectedItems - The currently selected items.
+ * @param onChange - Callback function to handle changes in selected items.
+ * @param onSubmit - Callback function to handle form submission.
+ * @param onCancel - Callback function to handle cancellation.
+ * 
+ * @returns The CheckBoxList component.
+ */
+export default function CheckBoxList({ title, items, selectedItems, onChange, onSubmit, onCancel }: CheckBoxListProps) {
+  const translations = useTranslations('CustomCheckBoxList');
+
+  const handleItemChange = (checked: boolean, name: string) => {
+    if (checked) {
+      onChange([...selectedItems, name]);
+    } else {
+      onChange(selectedItems.filter(item=> item !==name));
+    }
+  };
+  
+  const handleAllChange = (checked: boolean) => {
+    onChange(checked ? items : []);
+  };
+  
+  const areAllSelected = items.length > 0 && selectedItems.length === items.length;
+    
+  return (
+    <form className={styles.filterInputContainer} onSubmit={onSubmit}>
+      <p>{title}</p>
+      <CheckboxGroup
+        className={styles.checkBoxList}
+      >
+        <Checkbox defaultChecked={areAllSelected}
+          id="checkbox-all"
+          labelText="All"
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => handleAllChange(event.target.checked)} />
+        {items.map((name: string) => (
+          <Checkbox
+            key={name}
+            id={`checkbox-${name}`}
+            labelText={name.charAt(0).toUpperCase() + name.slice(1)}
+            onChange={(event: React.ChangeEvent<HTMLInputElement>) => handleItemChange(event.target.checked, name)}
+            checked={selectedItems.includes(name)}
+          />
+        ))}
+      </CheckboxGroup>
+      <div className={styles.buttonContainer}>
+        <Button type="button" kind="secondary" onClick={onCancel}>{translations('cancel')}</Button>
```

Comment:
Trying this out locally - the "cancel" button doesn't seem to do anything for some of the input fields like requestor and test run name?

Edit: Ah, it seems to revert back to the saved changes which makes sense, though going back to clear criteria might be tricky. I wonder if we need a control to reset the query back to its initial state?